### PR TITLE
Use github.sha instead of github.ref in reusable workflows

### PR DESCRIPTION
**Pull request by mazhelez**

### CODE REVIEW

File: Templates/AppSource App/.github/workflows/PullRequestHandler.yaml [hunk 1]
```diff
@@ -93,7 +93,6 @@ jobs:
     with:
       shell: ${{ needs.Initialization.outputs.githubRunnerShell }}
       runsOn: ${{ needs.Initialization.outputs.githubRunner }}
-      checkoutRef: refs/pull/${{ github.event.number }}/merge
```
Comment (freddydk) [inline Templates/AppSource App/.github/workflows/PullRequestHandler.yaml:96]: Is this really correct???

Comment (freddydk) [reply Templates/AppSource App/.github/workflows/PullRequestHandler.yaml:96]: I think we went back and forth a few times back then to find this

File: Templates/AppSource App/.github/workflows/PullRequestHandler.yaml [hunk 2]
```diff
@@ -38,6 +38,7 @@ jobs:
       telemetryScopeJson: ${{ steps.init.outputs.telemetryScopeJson }}
       githubRunner: ${{ steps.ReadSettings.outputs.GitHubRunnerJson }}
       githubRunnerShell: ${{ steps.ReadSettings.outputs.GitHubRunnerShell }}
+      checkoutRef: ${{ github.event.pull_request.merge_commit_sha }}
       projects: ${{ steps.determineProjectsToBuild.outputs.ProjectsJson }}
```
Comment (freddydk) [inline Templates/AppSource App/.github/workflows/PullRequestHandler.yaml:42]: I think you placed this in the wrong location (under outputs) - should be a parameter for _BuildALGo...

**Review by freddydk — CHANGES_REQUESTED** [review]

File: Templates/AppSource App/.github/workflows/PullRequestHandler.yaml (hunk 2 above)
Comment (mazhelez) [reply Templates/AppSource App/.github/workflows/PullRequestHandler.yaml]: Yes, yes... figured it out the long way 😄

**Comment by freddydk** [comment]
I guess this should also resolve the bug we have with PR handler fails if the PR is merged before completing - or was this already resolved earlier?

**Review by freddydk — APPROVED** [review]

**Review by aholstrup1 — APPROVED** [review]

**Pull request merged**